plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMapping.java

```diff
@@ -84,7 +87,9 @@ public S3SecurityMapping(
         requireNonNull(accessKey, "accessKey is null");
         requireNonNull(secretKey, "secretKey is null");
         checkArgument(accessKey.isPresent() == secretKey.isPresent(), "accessKey and secretKey must be provided together");
-        this.credentials = accessKey.map(access -> new BasicAWSCredentials(access, secretKey.get()));
+        this.credentials = sessionToken.isEmpty() ?
```

Comment:
validate that if `sessionToken` is provided then we require also `accessKey` and `secretKey` to be provided.

Reply:
done